### PR TITLE
Updates FPS Setting

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
This updates the default framerate for new players and adds a better description.

The default setting should be the best setting for most users. Why 60 specifically?
-Most users likely have a 60hz Monitor, and 60 is devisible by 20 ticks so ideally every tick frametime should be as  stable as possible. 
-From my measurements 60 Fps Frametime seems more stable than 20 Fps. 
-60 FPS to avoid frame tearing.
-Byond seems to only pump out about 64 FPS Max.

20 FPS
https://drive.google.com/uc?id=1kAu4ngM29c2w7U8XKsc6LDepFIjFVOy7

60 FPS
https://drive.google.com/uc?id=1DlKAL3qxu8gYaiDKbTUCzoojA9cyhlZe

20 FPS Jumping test
https://drive.google.com/uc?id=1OptrWMXsPVY7RQtIiyiYEc30FRRoIti2

60 FPS Jumping test
https://drive.google.com/uc?id=1ustaQB2RIDr0U45I7QQVZoAupJ0BIa-g

Please Vote after watching all 4 videos!

## Why It's Good For The Game

I personally think the difference is day and night - This will make the game more attractive to new players and more enjoyable to users who didn't manually specify the 20 FPS limit.

I think this change is important since I and I'm sure many other users saw the fps_max = 0 setting in-game and assumed it meant 'uncapped' as it does in every other game. To be honest I played like this for years.
## Changelog
:cl:
tweak: Default Framerate is now 60 FPS.
tweak: 0 now gets changed to the actual framerate in the menu.
tweak: Framerate menu gives a better description and shows the proper number.
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
Glide size for any client fps other than 20 (world.fps) is straight up bugged and it's extremely visible once you notice it.
It's most noticeable when you stutterstep or run around corners.
http://www.byond.com/forum/post/2241289 for reference.

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -1537,10 +1537,13 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 						preferred_map = maplist[pickedmap]
 
 				if ("clientfps")
-					var/desiredfps = input(user, "Choose your desired fps. (0 = synced with server tick rate (currently:[world.fps]))", "Character Preference", clientfps)  as null|num
+					var/desiredfps = input(user, "Please type in your Monitor Refresh Rate. Minimum Framerate is [world.fps])", "Character Preference", clientfps)  as null|num
```
Comment (nemvar) [inline code/modules/client/preferences.dm:1540]: Minimum Framerate isn't actually 20. It's entirely possible to set it to any value between 1-19 currently. If someone wants to ruin their experience with that, let them.
I also disagree that the current message isn't informative and with sanitizing the value with something that isn't a constant.
Just add a line in the changelog that tell people to try out 60 fps.

In short, I liked your previous PR better and would happily slam the merge button on it.

**Review by nemvar — APPROVED** [review]
Disregard my previous comment. I was working an outdated version. Did some testing and this actually works fine now.

> To be honest I played like this for years.

Good for you. This has been bugged in horrendous ways up until midway into last month.

**Comment by MrStonedOne** [comment]
40 and 100 both look better on 60fps monitors then 60 does.

byond rounds this to a whole millisecond. so 60fps isn't even a real thing, and the actual value it uses (62.5 fps) is out of sync with the server.

This is also why I wouldn't attempt to abstract this away to "monitor refresh rate" people know what the fuck a fps is.

**Comment by MrStonedOne** [comment]
http://www.byond.com/forum/post/2127554 for some discussion on this.

**Review by MrsTonedOne — CHANGES_REQUESTED** [review]
Because the best fps is one that is a multiple or has a careful relation to the server's game tick, and that number is closely related to how the server is configured: the default client fps should be made either a config, or a multiple of the server's tick rate. 

The former works better because of how byond rounds these values and how that will influence what client fps is best for any given server.

Further more, there must be some way to distinguish between players still using the default fps, and players that have explicitly set one, so that a server looking to change their tick rate can have most clients use a new default client fps. `null` or `-1` (with a define) for the preference datum's var will work, as long as that doesn't make it to the client's fps var.

The config recommended default client fps should either be set to 40 or 100, depending on which turns out to be better on testing. I'm thinking 40 might be more sensible.

Later we can work on improving automatic client fps, with things to detect the performance of the machine.

**Pull request closed without merging**